PicPi/AddPhotoViewController.swift

```diff
@@ -87,17 +110,50 @@ class AddPhotoViewController: UIViewController {
       ]
     )
   }
+
+  private func updateUI() {
+    tableView.reloadData()
+    updateSendButton()
+  }
+
+  private func updateSendButton() {
+    let isPhotosAndPictureFrameAlreadySelected =
+      !(photoViewModel.selectedAssets.isEmpty || photoViewModel.selectedPictureFrames.isEmpty)
+    sendButton.isEnabled = isPhotosAndPictureFrameAlreadySelected
+  }
+
+  var currentSession: MediaUploaderSession?
   
   // MARK: - Buttons Actions
   @objc func sendButtonPressed(sender: UIButton!) {
-//    photoUploadManager.startUploadPreviouslySelectedPhotos()
-//    sendButton.isEnabled = false
-    photoUploadManager.getThumbnails()
+    sendButton.isEnabled = false
+    let session = photoViewModel.beginUpload()
+    session.didCompleteSession.add(self) { [weak self] success in
+      guard let self = self else { return }
+
+      let message = success ? "Success!" : "something went wrong"
+      let alert = UIAlertController(title: "Upload", message: message, preferredStyle: .alert)
+
+      if success {
+        alert.addAction(UIAlertAction(title: "Done", style: .default, handler: { _ in
+          self.presentingViewController?.dismiss(animated: true)
+        }))
+      } else {
+        alert.addAction(UIAlertAction(title: "Dismiss", style: .default, handler: { _ in
+          self.sendButton.isEnabled = true
+        }))
+      }
+
+      self.present(alert, animated: true)
+    }
```

Comment:
Any thoughts of creating struct for Alerts , and create Alert Presenter?

Reply:
I've made extensions on UIAlertController that helped created alerts but I'm open to having a struct for alerts

Reply:
Spoke offline, flipped a coin and going with extensions